src/axom/core/examples/core_containers.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*! \file core_containers.cpp
+ *  \brief This example code is a demonstration of the Axom Core containers.
+ */
+
+ /* This example code contains snippets used in the Core Sphinx documentation.
+  * They begin and end with comments such as
+  *
+  * timer_start
+  * timer_end
+  *
+  * each prepended with an underscore.
+  */
+
+
+// Axom includes
+#include "axom/core/Array.hpp"
+#include "axom/core/memory_management.hpp"
+
+#ifdef WIN32
+#include "windows.h"
+void sleep(int numSeconds)
+{
+  int numMilliSecs = numSeconds * 1000;
+  Sleep(numMilliSecs);
+}
+#else
+#include <unistd.h> // for sleep()
+#endif
+
+// C/C++ includes
+#include <iostream>
+#include <vector>
+
+void showArray(axom::Array<int> &a, const char* name)
+{
+  std::cout << "Array " << name << " = [";
+  for (int i = 0; i < a.size(); ++i)
+  {
+    if (i > 0) { std::cout << ", "; }
+    std::cout << a[i];
+  }
+  std::cout << "]" << std::endl;
+}
+
+void showTupleArray(axom::Array<int> &a, const char* name)
+{
+  std::cout << "Array " << name << " with " << a.size() << " " <<
+    a.numComponents() << "-tuples = [" << std::endl;
+  for (int i = 0; i < a.size(); ++i)
+  {
+    std::cout << "  ["; 
+    for (int j = 0; j < a.numComponents(); ++j)
+    {
+      if (j > 0) { std::cout << ", "; }
+      std::cout << a(i, j);
+    }
+    std::cout << "]" << std::endl;
+  }
+  std::cout << "]" << std::endl;
+}
+
+void demoArrayBasic()
+{
+  // _arraybasic_start
+  // Here is an Array of ints with length three.
+  axom::Array< int > a(3);
```

Comment:
Heads up @bmhan12 -- this will have to be updated when you get back to your array branch.